hooks/use-upload-file.ts

```diff
@@ -39,8 +40,16 @@ export function useUploadFile({
           process.env.NEXT_PUBLIC_CLOUDINARY_UPLOAD_PRESET || "",
         );
 
+        let resource: string;
+        if (file.type.startsWith("image/")) {
+          resource = "image/upload";
```

Comment:
Why not setting the resource variable to just "image", "video" or "raw" and changing the url to /${resource}/upload?
```suggestion
          resource = "image";
```

Reply:
since all of them end with upload we can do it